src/main.rs

```diff
@@ -0,0 +1,78 @@
+                  .subcommand(SubCommand::with_name("create")
+                                .about("create a new motion using the template")
+                                .arg_from_usage("<name> 'the name to use for the new motion'"))
+                  .subcommand(SubCommand::with_name("add")
+                                .about("adds n motions")
+                                .arg_from_usage("[n] 'how many motions to add, defaults to 1'"))
+                  .subcommand(SubCommand::with_name("sub")
+                                .about("subtracts n motions")
+                                .arg_from_usage("[n] 'how many motions to substract, defaults to 1'"))
+                  .subcommand(SubCommand::with_name("shift")
+                                .about("goes to the nth motion relative to the current motion")
+                                .arg_from_usage("--n=<n> 'the amount of motions to move relative to the current one,
+                                                 must be given as a number in the form of --n=[number]'"))
+                  .subcommand(SubCommand::with_name("goto")
+                                .about("go to the nth motion")
+                                .arg_from_usage("<n> 'the motion to go to, 0 based'"))
+                  .get_matches();
+  // Get all the specified variables or set them to their default values
+  let target = matches.value_of("url").unwrap().to_string();
+  let directory = matches.value_of("path").unwrap_or(".").to_string();
+  let mots = motions::discover(&directory);
+  // TODO Adquire driver properly!
+  let mut driver = get_driver(target);
+  // Go through and find what matched
+  match matches.subcommand() {
+    ("ls", Some(_)) => ls(directory, mots),
+    ("up", Some(_)) => accelerator::up(&mut driver, &mots),
+    ("down", Some(_)) => accelerator::down(&mut driver, &mots),
+    ("redo", Some(_)) => accelerator::redo(&mut driver, &mots),
+    ("reset", Some(_)) => accelerator::reset(&mut driver, &mots),
+    ("create", Some(m)) => create(directory, value_t_or_exit!(m.value_of("name"), String)),
+    ("add", Some(m)) => accelerator::shift(&mut driver, &mots, value_t!(m.value_of("n"), isize).unwrap_or(1)),
+    ("sub", Some(m)) => accelerator::shift(&mut driver, &mots, value_t!(m.value_of("n"), isize).unwrap_or(1) * -1),
+    ("shift", Some(m)) => accelerator::shift(&mut driver, &mots, value_t_or_exit!(m.value_of("n"), isize)),
+    ("goto", Some(m)) => accelerator::goto(&mut driver, &mots, value_t_or_exit!(m.value_of("n"), isize)),
+    _ => println!("Nothing to do!\nRe-run with --help for more information"),
+  }
+}
+
+fn get_driver(target: String) -> Box<Driver> { Box::new(drivers::DefaultDriver::new(target)) }
+
+fn ls(dir: String, mots: Vec<Motion>) {
+  println!("{} contains: {} motions\n", dir, mots.len());
+  for mot in mots {
+    println!("{}", mot.name);
+  }
+}
+
+fn create(directory: String, name: String) { motions::create(directory, name); }
```

Comment:
Could you expand this function into three lines? Generally that‘s good practice for readability and code coverage tests.